### PR TITLE
x86isa -- updates to asmtest and implement cmpxchg8b/16b instructions

### PR DESCRIPTION
**Pull request by solswords**

### CODE REVIEW

File: books/projects/x86isa/machine/instructions/exchange.lisp [hunk 1]
```diff
@@ -363,3 +363,98 @@
     x86))
 
 ;; ======================================================================
+;; INSTRUCTION: CMPXCHG8B/16B
+;; ======================================================================
+
+(def-inst x86-cmpxchg8b/16b
+
+  ;; Op/En: MR
```
Comment (acoglio) [inline books/projects/x86isa/machine/instructions/exchange.lisp:371]:
```suggestion
  ;; Op/En: M
```

**Review by acoglio — APPROVED** [review]
I just left a comment/edit/suggestion for a typo, which you can accept by clicking on it (it will create a commit).

File: books/projects/x86isa/machine/inst-listing.lisp [hunk 2]
```diff
@@ -9752,7 +9752,7 @@
               :GROUP '(:GROUP-9)
               :FEAT '(:CMPXCHG16B))
           (ARG :OP1 '(M DQ))
-          'NIL
+          '(x86-cmpxchg8b/16b)
```
Comment (yaso9) [inline books/projects/x86isa/machine/inst-listing.lisp:9755]:
```suggestion
          '(X86-CMPXCHG8B/16B)
```

File: books/projects/x86isa/machine/inst-listing.lisp [hunk 3]
```diff
@@ -9765,7 +9765,7 @@
               :GROUP '(:GROUP-9)
               :FEAT '(:CMPXCHG16B))
           (ARG :OP1 '(M Q))
-          'NIL
+          '(x86-cmpxchg8b/16b)
```
Comment (yaso9) [inline books/projects/x86isa/machine/inst-listing.lisp:9768]:
```suggestion
          '(X86-CMPXCHG8B/16B)
```

**Review by yaso9 — APPROVED** [review]

**Review by ericwhitmansmith — APPROVED** [review]
All Kestrel proofs still work after these changes.

**Pull request merged**